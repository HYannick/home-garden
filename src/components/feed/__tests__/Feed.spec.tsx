import React from 'react';
import 'jest-dom/extend-expect';
import { cleanup } from '@testing-library/react';
import { renderWithTheme } from '../../../theme-wrapper';
import FeedView, { FeedViewProps } from '../Feed.view';

afterEach(cleanup);
describe('PlantList', () => {
  const feedControls = [
    {
      key: 1,
      icon: () => <div>Right</div>,
      disabled: false,
      onClick: jest.fn(),
    },
    {
      key: 2,
      icon: () => <div>Left</div>,
      disabled: false,
      onClick: jest.fn(),
    },
  ];

  const defaultProps: FeedViewProps = {
    t: jest.fn(),
    fadeTransition: [],
    articleTransition: [],
    feedControls,
    articles: [{
      id: 1,
      link: 'green-garden.com',
      picture: 'draecena.jpg',
      title: 'How to be a hero',
      source: 'green-hero.com',
    }],
    hasErrors: false,
  };
  it('should render properly', () => {
    const { container } = renderWithTheme(
      <FeedView {...defaultProps}/>,
    );
    expect(container).toMatchSnapshot();
  });
});
