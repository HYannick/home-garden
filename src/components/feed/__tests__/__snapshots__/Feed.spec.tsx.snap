// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlantList should render properly 1`] = `
<div>
  <div
    class="css-1pca6wa"
  >
    <div
      class="css-1u8bncc"
    >
      <div
        class="css-30uqkx"
      >
        <div>
          <h1
            class="css-21eb7x"
          />
          <h6
            class="css-og9iw5"
          />
        </div>
        <div
          class="css-1ytm51z"
        >
          <button
            class="css-1b1pyeo"
          >
            <div>
              Right
            </div>
          </button>
          <button
            class="css-1b1pyeo"
          >
            <div>
              Left
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      class="css-kps18y"
    />
  </div>
</div>
`;
